src/hotspot/share/services/threadService.cpp

```diff
@@ -167,6 +170,7 @@ void ThreadService::remove_thread(JavaThread* thread, bool daemon) {
     // We did not get here via JavaThread::exit() so current_thread_exiting()
     // was not called, e.g., JavaThread::cleanup_failed_attach_current_thread().
     decrement_thread_counts(thread, daemon);
+    ThreadService::incr_exited_allocated_bytes(thread->cooked_allocated_bytes());
```

Comment:
By doing this here you increase the likelihood of double-accounting for this thread. If you do this after the thread is no longer on any threads-list you may miss its contribution entirely, but you won't double-count it.

Reply:
If the `thread` is no longer on any ThreadsList, then it is not safe to access `thread`.

Reply:
@dcubed-ojdk This is the current thread acting on itself

Reply:
This method (i.e. `ThreadService::remove_thread()`) is called from `Threads::remove()` *after* the thread was removed from the thread list:
```cpp
void Threads::remove(JavaThread* p, bool is_daemon) {
  ...
    // Maintain fast thread list
    ThreadsSMRSupport::remove_thread(p);
    ...
    ThreadService::remove_thread(p, is_daemon);
```

But if we reach here from `JavaThread::cleanup_failed_attach_current_thread()` as the comment implies (`JavaThread::cleanup_failed_attach_current_thread()` calls `Threads::remove()` in the case of an attach failure), calling `ThreadService::incr_exited_allocated_bytes()` is probably irrelevant, because a thread which failed to attach can't allocate anyway.

Reply:
Done.